### PR TITLE
24502 - add modal for invalid name request and re-organize error page

### PR DESCRIPTION
**Pull request by patrickpeinanw**
*Issue:*https://github.com/bcgov/entity/issues/24502

*Description of changes:*
Now the error page has three different scenarios:
  1) auth error for a business 
  2) auth error for a bootstrap business
  3) invalid name request

Note: in Filing UI, the header for Invalid Name Request is always "Invalid Incorporation Application", in the new dashboard, the dialog header is "Invalid Name Request". 

Screenshots for different types of invalid name request (taken from the cypress test)
image
image
image
image
image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-dashboard-dev--pr-90-un4amkde.web.app

File: src/pages/errors.vue [hunk 1]
```diff
@@ -36,49 +40,106 @@ const defaultDialogOptions = {
...
-    :display="showUnauthorizedAccess"
+    :display="showUnauthorizedAccess || !!invalidNameRequest"
     :options="defaultDialogOptions"
     @close="goHome()"
   >
     <template #header>
       <span class="text-2xl font-light">
-        Unable to Access Business
+        <template v-if="showUnauthorizedAccess">
+          {{ isTempReg ? 'Unable to Access Incorporation Application' : 'Unable to Access Business' }}
```
Comment (severinbeauvais) [inline src/pages/errors.vue:77]: A Registration, an Amalgamation, and a Continuation In also use a temporary registration number. You probably need a different message for those.

Or, maybe you can get away with "Unable to Access Registration", but confirm with your UX person.

File: src/pages/errors.vue [hunk 2]
```diff
@@ -36,49 +40,106 @@ const defaultDialogOptions = {
...
+          </templale>
+        </template>
+        <!-- Invalid Name Request Dialog -->
+        <template v-else>
+          <p class="font-15">
+            {{ nrErrorMessage }}
+          </p>
+          <p class="mt-4">
+            You can retry now, or you can exit and try to access this Incorporation Application
+            at another time.
```
Comment (severinbeauvais) [inline src/pages/errors.vue:130]: The Retry button should (but doesn't) reload all the data to see if it's now OK (like Filings UI does).

File: src/pages/errors.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/pages/errors.vue]: Sample sole prop registration with expired NR:
https://business-dashboard-dev--pr-90-un4amkde.web.app/TFyrdiG0Vg

Still shows "Invalid Incorporation Application":

image

File: src/pages/errors.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/pages/errors.vue]: This is still all that happens on Retry:

image

File: src/pages/errors.vue (hunk 1 above)
Comment (patrickpeinanw) [reply src/pages/errors.vue]: The header should now show "Invalid Name Request". I just ran gcbrun. It may take some time for the temp url to be updated.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-dashboard-dev--pr-90-un4amkde.web.app

**Review by hfekete — APPROVED** [review]
Nice work, looks good to me!
Lets see if issues Severin found are resolved.

File: src/pages/errors.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/pages/errors.vue]: Looks good. Resolving.

Comment (severinbeauvais) [reply src/pages/errors.vue]: Tests:

image

image

image

image

File: src/pages/errors.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/pages/errors.vue]: Here are the network calls on Retry. It looks like this is enough to see if anything has changed (and therefore if it's worth retrying).

image

Resolving.

**Pull request merged**